cilium-cli/connectivity/tests/errors.go

```diff
@@ -480,3 +487,18 @@ var (
 	// ccgAlphaResourceDeprecation is the same as bgpAlphaResourceDeprecation but for the CiliumCIDRGroup.
 	ccgAlphaResourceDeprecation = regexMatcher{regexp.MustCompile(`cilium.io/v2alpha1 CiliumCIDRGroup is deprecated`)}
 )
+
+// hasLegitimateOperatorRestartPatterns returns true only for intentional, benign
+// operator restarts. Consider legitimate only if the leader handover message
+// is present AND there are no error indicators in the same logs.
+func hasLegitimateOperatorRestartPatterns(logs []byte) bool {
+	logStr := string(logs)
+	if !strings.Contains(logStr, "Leader election lost") {
+		return false
+	}
+	// Check for error-level logs along with the leader election lost message.
+	if strings.Contains(logStr, "level=error") || strings.Contains(logStr, "[error]") || strings.Contains(logStr, "[critical]") {
+		return false
+	}
```

Comment:
This seems like the newer changes which are trying to duplicate existing error detection logic in `findUniqueFailures()`, which will cause weird behavior because sometimes errors won't be ignored as expected per that function. I don't think these changes are quite right.
```suggestion
```